### ISSUE TITLE
Incorrect project-level roles shown in project settings

### ISSUE THREAD

**Issue by rolodato** [SolidStateGroup/bullet-train-api#4924]
### How are you running Flagsmith

- [ ] Self Hosted with Docker
- [ ] Self Hosted with Kubernetes
- [x] SaaS at flagsmith.com
- [ ] Some other way (add details in description below)

### Describe the bug

The list of project-specific roles in the Project Settings page only seems to show 1 role, even if other roles should be shown. It can also incorrectly show roles if they used to be specific to a project in the past.

### Steps To Reproduce

1. Create two roles
2. Give one role Project Admin permissions to any project
3. Go to Project Settings > Permissions > Roles, and validate that that role is present
4. Give the other role Project Admin permissions to the same project
5. Go to the project roles page again - only the role from step 2 will be shown

If you remove Project Admin permissions from the first role, it will still be shown incorrectly in the project roles page.

If you delete the first role, the second role will now be shown correctly in the project roles page.

The endpoint returning incorrect data is `GET /api/v1/organisations/15467/roles/717/projects-permissions/?project=22470`

### Expected behavior

All project roles should be returned in the project settings page.

### Screenshots

https://github.com/user-attachments/assets/d5c0afbd-cd04-41eb-92f7-0d75e7b6cd19

**Comment by gagantrivedi** [comment]
@rolodato  what is wrong with the data returned by `GET /api/v1/organisations/15467/roles/717/projects-permissions/?project=22470` endpoint?

From what I can see, UI is not correctly picking up the roles that it needs to show on that page @kyle-ssg have we fixed this and environment settings page?

**Comment by kyle-ssg** [comment]
@gagantrivedi I think this is a duplicate of https://github.com/Flagsmith/flagsmith/issues/3784 in that every user should see the full list of roles.

There's nothing wrong with the endpoint you mentioned it's just the role is not coming back in the list i.e ``https://api.flagsmith.com/api/v1/organisations/15467/roles/`` is not returning it. 

Assign back to me if you don't think the above is the case but it seems that any user should be able to get the list of roles.

**Comment by rolodato** [comment]
`GET /api/v1/organisations/15467/roles/` returns two roles, which both give project admin access to the same project:

```
{
  "count": 2,
  "next": null,
  "previous": null,
  "results": [
    {
      "id": 719,
      "name": "Test Project Admins 2",
      "description": "",
      "organisation": 15467,
      "tags": []
    },
    {
      "id": 727,
      "name": "admins",
      "description": "",
      "organisation": 15467,
      "tags": []
    }
  ]
}
```

However `GET /api/v1/organisations/15467/roles/719/projects-permissions/?project=22470` only returns the first one:

```
{
  "count": 1,
  "next": null,
  "previous": null,
  "results": [
    {
      "id": 1097,
      "role": 719,
      "project": 22470,
      "permissions": [],
      "admin": true
    }
  ]
}
```

**Comment by gagantrivedi** [comment]
Yep, that's because the first endpoint returns all the roles that belong to an organization, while the second endpoint returns all the project permissions a role has

**Comment by rolodato** [comment]
Got it - my bad, I didn't notice that. In that case this seems like a FE issue where only the first role is being requested?

**Comment by gagantrivedi** [comment]
> Got it - my bad, I didn't notice that. In that case this seems like a FE issue where only the first role is being requested?

Yep, looks like that cc @kyle-ssg

**Comment by gagantrivedi** [comment]
> [@gagantrivedi](https://github.com/gagantrivedi) I think this is a duplicate of [#3784](https://github.com/Flagsmith/flagsmith/issues/3784) in that every user should see the full list of roles.
> 
> There's nothing wrong with the endpoint you mentioned it's just the role is not coming back in the list i.e `https://api.flagsmith.com/api/v1/organisations/15467/roles/` is not returning it.
> 
> Assign back to me if you don't think the above is the case but it seems that any user should be able to get the list of roles.


Looks like this is not the case, the above endpoints returns both the roles

**Comment by kyle-ssg** [comment]
Ah ok yep I can replicate both the above and #3784, you're right they are separate issues.
